### DISCUSSION TITLE
Set text/printing darkness via code

### DISCUSSION THREAD

**Discussion by ciaransweet** [python-escpos/python-escpos#509] [Q&A] [answered]
Hey Folks,

Firstly, very thankful for this package, it worked out the box with my second hand TM-TM88IV and i've been having a blast!

I've replaced the serial interface with USB and I was wondering if there's a way to set the darkness of the prints via `python-escpos`?

I'm 99% sure when it had the serial interface, I was able to change these in the settings on the printer with the feed button, but those options seem to have gone.

Either way, it'd be cool if I could have different `.text()` calls have different 'darknesses'.

Thanks for any help!

**Comment by belono** [comment, marked as answer]
Hi @ciaranevans and thanks for your words!

The [escpos.set()](https://github.com/python-escpos/python-escpos/blob/master/src/escpos/escpos.py#L671) method has a `density` param ranging from 0 to 8. Is it what you are looking for?

Greetings!
